docs/data/toolpad/authentication.md

```diff
@@ -0,0 +1,84 @@
+# Authentication
+
+<p class="description">Toolpad provides working authentication out of the box, with support for multiple providers</p>
```

Comment:
Does it make sense to create initial core documentation as part of this PR? Feels like it needs more shaping first. Perhaps it makes sense to just do the rename of /toolpad => /toolpad/studio here to make space for core docs and set up those in a separate PR?